src/features/ProfilePage/components/AccountInfo.tsx

```diff
@@ -17,28 +22,27 @@ import { Section, Value } from '.';
 import Text from '@/components/Text';
 import { TextButton } from '@/components/Buttons';
 
-import type { UserRole } from '@/features/Authentication/authenticationApi';
-
-type Props = {
-  userRole: UserRole;
-};
-
-const AccountInfo = ({ userRole }: Props) => {
+const AccountInfo = () => {
   const { t } = useTranslation('profile');
-  const { loginName } = useAppSelector(selectUserInfo);
-
-  const isMentor = userRole === 'mentor';
+  const { id, login_name: loginName, role } = useAppSelector(selectAccount);
+  const isMentor = useAppSelector(selectIsMentor);
+  const [deleteAccount] = useDeleteAccountMutation();
+  const [logout] = useLogoutMutation();
+
+  const confirmDeletion = async () => {
+    try {
+      await deleteAccount(id).unwrap();
+      logout();
+    } catch (err) {
+      return;
+    }
+  };
```

Comment:
Instead of unwrapping, why not implement the onQueryStarted-thing?

```typescript
// here we have dispatch also as property! We can trigger actions with it.
 async onQueryStarted(_, { dispatch, queryFulfilled }) {
        try {
          await queryFulfilled;
          
          // Triger the logout mutation after deleteAccount is successful
          dispatch(apiSlice.endpoints.logout.initiate());
        } catch (err) {
```

Reply:
Wait a minute! I seem to have accidentally done both. 😅

Reply:
Now fixed!